networkit/cpp/graph/Graph.cpp

```diff
@@ -591,12 +591,20 @@ edgeweight Graph::weightedDegreeIn(node u, bool countSelfLoopsTwice) const {
 
 /** EDGE MODIFIERS **/
 
-void Graph::addEdge(node u, node v, edgeweight ew) {
+void Graph::addEdge(node u, node v, edgeweight ew, bool checkForMultiEdges) {
     assert(u < z);
     assert(exists[u]);
     assert(v < z);
     assert(exists[v]);
 
+    if (checkForMultiEdges) {
+        if (std::find(outEdges[u].begin(), outEdges[u].end(), v) != outEdges[u].end()
+            || std::find(outEdges[v].begin(), outEdges[v].end(), u) != outEdges[v].end()) {
```

Comment:
This check doesn't look right. If we have a directed graph with the edge `(v, u)`, then we cannot add the edge `(u, v)`.
Can't we simply use `if (hasEdge(u, v))`?

Reply:
True, that makes more sense.